### ISSUE TITLE
SA4006: false positive on unused variable

### ISSUE THREAD

**Issue by dsnet** [dominikh/go-staticcheck#25]
Consider the following:
```go
package foo

type R struct {buf []byte}

func (r *R) Read(buf []byte) (int, error) {
	var rdCnt int
	for len(buf) > 0 {
		if len(r.buf) > 0 {
			cpCnt := copy(buf, r.buf)
			buf, r.buf = buf[cpCnt:], r.buf[cpCnt:] // buf is unused (SA4006)
			rdCnt += cpCnt
			break
		}

		// Do work to populate r.buf
	}
	return rdCnt, nil
}
```

The tool reports:
```
foo.go:10:4: this value of buf is never used (SA4006)
```

However, this is wrong since assigning to buf is critical to the operation of the outer for loop.

**Comment by dsnet** [comment]
Just kidding! The assignment really is unused. Awesome.
